autora/theorist/bms/parallel.py

```diff
@@ -119,7 +119,7 @@ def tree_swap(self) -> Tuple[Optional[str], Optional[str]]:
             self.trees[self.Ts[nT2]] = t1
             t1.BT = BT2
             t2.BT = BT1
-            self.t1 = self.trees["1"]
+            self.t1 = self.trees["1.0"]
```

Comment:
what does the 1.0 do?

Reply:
It's probably easier to see looking in the code but I'll do my best to describe it. The keys of the dictionary of temperatures are created as numerical values and then recast as strings. Originally the code contained one integer ```1``` and the rest floats. When we type casted, we had to change an integer to a float. ```1``` became ```1.0```, which is different when cast as a string and so is a distinct key value. The code assumes a one-to-one correspondence between temperature values and dictionary keys. The code also hard codes one tree with ```1``` and then iteratively creates trees from the temperatures, but skips ```1``` (which we changed to ```1.0```). So the parallel machine scientist held a tree for a temperature value of ```1``` and of ```1.0```, and considers the the key ```"1"``` to hold the best model and ignores ```"1.0"```. When comparing trees in ```tree_swap()```, it only chooses among those that correspond to a temperature value stored in, which are recorded as floats, and so ignores ```"1"```. Hence the model selected by BMS is unaffected by ```tree_swap()```. The tree corresponding to ```"1"``` does improve over time because it is still affected by ```mcmc_step()```. However, we lose the functionality of having higher temperatures, tree swaps, and the other 95% of the training results.

Reply:
Tl;dr: when we type set temperatures to be floats, the hooks forced us to change a ```1``` to a ```1.0```. Turns out there was some hard coding which relied on it being specifically ```1```.